app/controllers/songs_controller.rb

```diff
@@ -0,0 +1,46 @@
+class SongsController < ApplicationController
+
+  def index
+
+  end
+
+  def new
+
+  end
+
+  def show
+    client = Grooveshark::Client.new({session: session[:groove_session]})
+    @groove_song = Song.find(params[:id])
+    @groove_url = client.get_song_url_by_id(@groove_song.link)
+  end
+
+  def create
+    song = Song.new(title: params[:song_name], artist: params[:song_artist], link: params[:song_id], playlist_id: params[:playlist_id], coverart: params[:coverart])
+    
+    if song.coverart != ''
+      song.coverart = "http://images.gs-cdn.net/static/albums/#{params[:coverart]}"
+      song.save
+    else
+      song.coverart = "http://1.bp.blogspot.com/-NFIeRN1TNpU/Ukou19njwHI/AAAAAAAAARQ/iypdhkQVZvI/s200/7313935-heavy-metal-rock-and-roll-devil-horns-hand-sign-with-a-black-leather-studded-bracelet.jpg"
+      song.save
```

Comment:
a) Setting the coverart should be the responsibility of the model.  Is `Song` coming from GrooveShark itself?  If so, you can still create your own Song model, which holds an instance of the groove_shark class and delegates methods to it.
b) It's risky to use an image that you are just pulling from blogspot as the default image.  Just download an image and store it in `app/assets/images`.  Check out [The Noun Project](http://thenounproject.com/).